include/nbl/asset/IPreHashed.h

```diff
@@ -41,36 +41,31 @@ class IPreHashed : public IAsset
 
 		static inline void discardDependantsContents(const std::span<IAsset*> roots)
 		{
-			struct stack_entry_t
-			{
-				const IAsset* asset;
-				core::unordered_set<const IAsset*> unvisitedChilds;
-			};
-			core::stack<stack_entry_t> stack;
-			core::unordered_set<const IAsset*> alreadyVisited;
-			auto push = [&stack,&alreadyVisited](const IAsset* node) -> void
+			core::stack<IAsset*> stack;
+			core::unordered_set<IAsset*> alreadyVisited; // whether we have push the node to the stack
+			core::unordered_set<IAsset*> alreadyDescended; // whether we have push the children to the stack
+			auto push = [&stack,&alreadyVisited](IAsset* node) -> void
 			{
 				if (!node)
 					return;
```

Comment:
you can assert this, `computeDependants` should never return you a nullptr (it has an obligation to skip nullptr potential deps)